float64 allFinger
float64 doubleFinger
float64 singleFinger